generators/typescript-v2/ast/src/ast/Function.ts

```diff
@@ -0,0 +1,89 @@
+import { CodeBlock } from "./CodeBlock";
+import { Comment } from "./Comment";
+import { Parameter } from "./Parameter";
+import { Type } from "./Type";
+import { AstNode } from "./core/AstNode";
+import { Writer } from "./core/Writer";
+
+export declare namespace Function {
+    interface Args {
+        name: string;
+        parameters: Parameter[];
+        return_: Type[];
+        body: CodeBlock | undefined;
+        docs: string | undefined;
+    }
+}
+
+export class Function extends AstNode {
+    protected _name: string;
+    protected _parameters: Parameter[];
+    protected _return_: Type[];
+    protected _body: CodeBlock | undefined;
+    protected _docs: string | undefined;
```

Comment:
nit: We can drop the `_` prefix on these parameters. We use the `_` suffix for keywords.